### ISSUE TITLE
Reference clustering - Nucmer error

### ISSUE THREAD

**Issue by niexiaoqing** [CSB5/OPERA-MS#63]
![image](https://user-images.githubusercontent.com/23168361/121625282-be890f00-caa5-11eb-9296-0cdc45524442.png)

**Comment by jsgounot** [comment]
Hi,

thanks for using Opera-MS. Are you able to access the file with a terminal ?

Regards,
jsgounot

**Comment by niexiaoqing** [comment]
> Hi,
> 
> thanks for using Opera-MS. Are you able to access the file with a terminal ?
> 
> Regards,
> jsgounot

no，nothing find in the dir

**Comment by Unaimend** [comment]
I have the same problem. But its seems that I am able to open the file with for example vim, but its not in temp folder but instead in the NUCMER_OUT folder.


`/scratch/SlurmTMP/sukem127.2123121/OPERA-MS/scratch/SlurmTMP/sukem127.2123121/RESULTS/intermediate_files/reference_clustering/NUCMER_OUT/temp_genome/119591-vs-121716_0.fa`
`
Check your paths and file permissions and try again
`

But at least for me the temp_genome folder is empty

**Comment by jsgounot** [comment]
Hi, can you past the last part of the logs ?
Which Opera-MS options did you used ?

**Comment by niexiaoqing** [comment]
> I have the same problem. But its seems that I am able to open the file with for example vim, but its not in temp folder but instead in the NUCMER_OUT folder.
> 
> `/scratch/SlurmTMP/sukem127.2123121/OPERA-MS/scratch/SlurmTMP/sukem127.2123121/RESULTS/intermediate_files/reference_clustering/NUCMER_OUT/temp_genome/119591-vs-121716_0.fa`
> `Check your paths and file permissions and try again`
> 
> But at least for me the temp_genome folder is empty

me too, the same problem

**Comment by niexiaoqing** [comment]
> Hi, can you past the last part of the logs ?
> Which Opera-MS options did you used ?

in intermediate_files/reference_clustering/NUCMER_OUT/log.txt
![image](https://user-images.githubusercontent.com/23168361/122322081-73647580-cf57-11eb-97cb-1ffcdd95a643.png)
in workdir there is a nucmer.error contains some error messages
![image](https://user-images.githubusercontent.com/23168361/122322226-ba526b00-cf57-11eb-9622-4f7c2e0ce3f4.png)
 Opera-MS options
![image](https://user-images.githubusercontent.com/23168361/122322324-e110a180-cf57-11eb-82c5-574058ce7904.png)

**Comment by niexiaoqing** [comment]
> > Hi, can you past the last part of the logs ?
> > Which Opera-MS options did you used ?
> 
> in intermediate_files/reference_clustering/NUCMER_OUT/log.txt
> ![image](https://user-images.githubusercontent.com/23168361/122322081-73647580-cf57-11eb-97cb-1ffcdd95a643.png)
> in workdir there is a nucmer.error contains some error messages
> ![image](https://user-images.githubusercontent.com/23168361/122322226-ba526b00-cf57-11eb-9622-4f7c2e0ce3f4.png)
> Opera-MS options
> ![image](https://user-images.githubusercontent.com/23168361/122322324-e110a180-cf57-11eb-82c5-574058ce7904.png)

and --contig-len-thr seems doesn't work

**Comment by jsgounot** [comment]
Hi, sorry, my schedule is a bit busy but I'm working on this issue, please don't close it.

**Comment by niexiaoqing** [comment]
> Hi, sorry, my schedule is a bit busy but I'm working on this issue, please don't close it.

i think i find why i got error message,maybe i should add --no-ref-clusting option.

**Comment by niexiaoqing** [comment]
> I have the same problem. But its seems that I am able to open the file with for example vim, but its not in temp folder but instead in the NUCMER_OUT folder.
> 
> `/scratch/SlurmTMP/sukem127.2123121/OPERA-MS/scratch/SlurmTMP/sukem127.2123121/RESULTS/intermediate_files/reference_clustering/NUCMER_OUT/temp_genome/119591-vs-121716_0.fa`
> `Check your paths and file permissions and try again`
> 
> But at least for me the temp_genome folder is empty

i think i find why i got error message,maybe i should add --no-ref-clusting option.

**Comment by Unaimend** [comment]
Yes, but I would like to have this ref/strain clustering OPERA-MS offers, that's one of the reasons I picked this assembler in my latest projects

**Comment by rhysnewell** [comment]
Look at the filepath that nucmer is trying to look at: `/scratch/SlurmTMP/sukem127.2123121/OPERA-MS/scratch/SlurmTMP/sukem127.2123121/RESULTS/intermediate_files/reference_clustering/NUCMER_OUT/temp_genome/119591-vs-121716_0.fa`
That file path does not exist, it's appending the directory where OPERA-MS is installed onto your working directory. It's wild.
OPERA-MS install: `/scratch/SlurmTMP/sukem127.2123121/OPERA-MS/`
Working dir: `/scratch/SlurmTMP/sukem127.2123121/RESULTS/intermediate_files/reference_clustering/NUCMER_OUT/temp_genome/119591-vs-121716_0.fa`

I've posted this in another thread, but thought I should post it here too so they can see it. Just change whatever code is messing with the file paths like that

**Comment by jsgounot** [comment]
Hi everyone. First, thanks to @rhysnewell for his comment on this thread. I just pushed a fix for this on github, please let me know if this does fix your problem. You don't need to recompile everything.

**Comment by Unaimend** [comment]
Sorry, I stopped working on this project, but I will try to redo it in the next few days and give you some feedback.
